cmake/kokkos_arch.cmake

```diff
@@ -111,7 +111,9 @@ IF (KOKKOS_ENABLE_CUDA_CONSTEXPR)
   ENDIF()
 ENDIF()
 
-IF (KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+IF(KOKKOS_ENABLE_SYCL)
+  SET(CUDA_ARCH_FLAG -Xsycl-target-backend -cuda-gpu-arch)
```

Comment:
This makes that code unreadable.  Put it elsewhere in a SYCL specific block with a comment

Reply:
I don't quite agree. This is the only place where we set `CUDA_ARCH_FLAG` so it makes sense to handle it here. Of course, I can just move
```
IF(KOKKOS_ENABLE_SYCL)
  SET(CUDA_ARCH_FLAG -Xsycl-target-backend -cuda-gpu-arch)
ENDIF()
```
directly after the existing `IF`-block and overwrite `CUDA_ARCH_FLAG` but that doesn't look better IMHO.

Reply:
Change it and make sure you comment.

Reply:
(Or find two reviewers that agree with you and think I am wrong)